### PR TITLE
Continuous idl

### PR DESCRIPTION
**Pull request by plehegar**
* Switch to continuous IDL support to facilitate linking
* Method description are rewritten as short algorithm to facilitate figuring out the values returned
* Improved introduction to "Extensions to Performance interface" section
* Some link cleanup
* other minor respec improvement

Hopefully, this is a step forward towards cleaning up the spec a bit but we'll need additional PRs to improve its coherence and consistency.
Merge with care, ie someone needs to be convinced that I didn't mess up the functions and that this is an improvement before we merge.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -294,59 +285,52 @@
...
...g/TR/html5/browsers.html#browsing-context">browsing</a> [[!HTML5]] or worker [[!WORKERS]] context's MUST be included as <a>PerformanceResourceTiming</a> objects in the <a href="https://www.w3.org/TR/performance-timeline-2/#sec-performance-timeline">Performance Timeline</a> of the relevant context, up to the <a>resource timing buffer size limit</a>. Resources that are retrieved from <a href="https://www.w3.org/TR/html5/browsers.html#relevant-application-cache">relevant application caches</a> or local resources MUST be included as <a>PerformanceResourceTiming</a> objects in the <a href="https://www.w3.org/TR/performance-timeline-2/#sec-performance-timeline">Performance Timeline</a> [[!PERFORMANCE-TIMELINE-2]]. Resources for which the <a title='fetch' href="https://www.w3.org/TR/html5/infrastructure.html#fetch">fetch</a> was initiated, but was later aborted (e.g. due to a network error) MUST be included as <a>PerformanceResourceTiming</a> objects in the <a href="https://www.w3.org/TR/performance-timeline-2/#sec-performance-timeline">Performance Timeline</a> and MUST contain initialized attribute values for processed substeps of the <a href="#processing-model">processing model</a>.</p>
```
Comment (igrigorik) [inline index.html]: > ... up to the <a>resource timing buffer size limit</a>. 

I think this clashes with PerfObserver: it sounds like observer should not get notifications if the buffer is full. I propose we drop that part of the sentence entirely from this section.

Also, for the last part on "initiated, but aborted".. we need to resolve https://github.com/w3c/resource-timing/issues/12.

File: index.html [hunk 2]
```diff
@@ -354,186 +338,198 @@
...
+    readonly        attribute DOMHighResTimeStamp responseEnd;
+    readonly        attribute unsigned long long  transferSize;
+    readonly        attribute unsigned long long  encodedBodySize;
+    readonly        attribute unsigned long long  decodedBodySize;
+    serializer = {inherit, attribute};
+};
+</pre>
+
+<!--
+<p dfn-for='PerformanceResourceTiming'></p>
```
Comment (igrigorik) [inline index.html]: drop?

File: index.html [hunk 3]
```diff
@@ -354,186 +338,198 @@
...
+when <a href="https://www.w3.org/TR/html5/infrastructure.html#fetch" title='fetch'>fetching</a> the resource and <strong>all</strong> the redirects or equivalent pass the <a>timing allow check</a> algorithm.</li>
+<li>zero, otherwise.</li>
+</ol>
+
+<p dfn-for='PerformanceResourceTiming'>
+On getting, the <dfn>fetchStart</dfn> attribute MUST return as follows:
+</p>
+<ol class='algorithm'>
+<li>
+The time immediately before the user agent starts to <a href="https://www.w3.org/TR/html5/infrastructure.html#fetch">fetch</a> the <em>final</em> resource in the redirection, if there are HTTP redirects or <a href="https://www.w3.org/TR/html5/infrastructure.html#concept-http-equivalent-codes" title='HTTP response codes equivalence'>equivalent</a>.
```
Comment (igrigorik) [inline index.html]: Aside: fwiw, I'm finding that pulling out common links into the footer and then using `[]`'s shorthand cleans up the spec quiet a bit and improves (editor) readability.

File: index.html [hunk 4]
```diff
@@ -41,6 +41,7 @@
     wg: "Web Performance Working Group",
     wgURI: "http://www.w3.org/2010/webperf/",
     wgPublicList: "public-web-perf",
+    subjectPrefix: "[ResourceTiming]",
```
Comment (igrigorik) [inline index.html]: FWIW, we should mark Arvind as no longer active.

**Comment by igrigorik** [comment]
Modulo a few minor knits, this is a huge improvement.. thanks for taking this on!

**Comment by plehegar** [comment]
Didn't use the []'s shorthand yet but could be dealt with separately.
#12 remains

**Comment by igrigorik** [comment]
After another careful read through it.. 👍  - merging.

I don't think we need to block this pull on #12.

**Pull request merged**